### ISSUE TITLE
BETTER Integration User Testing

### ISSUE THREAD

**Issue by macintoshpie** [SEED-platform/seed#2808]
## context
SEED has integrated BETTER analysis for running building and portfolio analyses. This ticket is intended to document these features for user testing.

## tasks

After setting up a SEED org with the required data (see Analysis requirements), you should be able to successfully verify the tests (see User Flows). Below is a sample file to help with testing.

[PM_BETTER_PORTFOLIO_INPUT.xlsx](https://github.com/SEED-platform/seed/files/6957710/PM_BETTER_PORTFOLIO_INPUT.xlsx)

#### Org
The organization must include a BETTER API token (#2779). Ask Ted or Austin for a test token.

#### Properties
For any property to successfully be included in an analysis, it must meet the following requirements:
- It must have these fields filled out:
  - City
  - State
  - Address Line 1
  - Gross Floor Area
  - Property Type (must be a valid property type according to ESPM, which will be the case if exported from there)
  - Property Name
- It must have linked meter data:
  - generally it should be electric grid or natural gas
  - there should be at least 12 consecutive monthly meter readings

### User Tests
The following tests are for after you've imported the necessary properties and meters.

Recommended configuration options for BETTER analysis (though you probably want to play around with different options)
<img width="597" alt="Screen Shot 2021-08-09 at 6 49 50 PM" src="https://user-images.githubusercontent.com/18518728/128783966-64c125c1-370d-4b18-83e3-487084bb688d.png">

1. Building Analysis: going to property detail page, then clicking "Run Analysis", choosing BETTER, creating the analysis, then running it after it's finished preparing. After successful completion, you should be able to go to the analysis results page and see the building analysis embedded in the page. You should be able to download the HTML file by clicking the html link in the table.
2. Portfolio analysis: Go to inventory list page, select multiple properties, click run analysis, choose BETTER, create the analysis (be sure to check "Portfolio analysis", then from org analysis page, wait for the analysis to finish preparing (change the page and come back to refresh -- this is really inconvenient for now), run the analysis. After successful completion, go to the analysis run page to view the results for an individual property. You should see both the building and portfolio analysis results embedded in the page. (#2783 )
3. If you violate one of the Analysis Requirements above, you should get a useful error message telling you what to fix when you try to include that property in an analysis. (e.g. remove city, property type, meters, etc) (#2772 )
4. If an org has an invalid BETTER API Token, SEED should let you know before it creates the analysis (#2779 )
5. If an org has no BETTER API token, SEED should let you know before it creates the analysis (#2779 )
6. After running an analysis (portfolio or individual), SEED should create additional "extra data" columns for the BETTER results. Go to the column list profile settings page, select all BETTER analysis columns, and you should see data in those columns from the inventory list page (#2775 )

**Comment by RDmitchell** [comment]
@macintoshpie -- I am having trouble importing an ESPM file with meter data into SEED.  We should probably talk about this on a call. Here is the folder where the file I am trying to import is: 

https://drive.google.com/drive/folders/1nKZuaUd3rAd9_r_lTvszpmDZkzMh3kKX?usp=sharing

**Comment by RDmitchell** [comment]
@macintoshpie -- The file (from JCC ESPM account) imported without issue except for 7 meters that didn't match to existing properties, which I will look into. The error I was having (520) was a Cloudflare problem because I was importing into an org with a LOT of properties. When I moved to a newer org with only a few properties, the file imported without issue.

**Comment by RDmitchell** [comment]
@macintoshpie / @aviveiros11 

I have tested ESPM to BETTER in SEED and it all seems to hang together. 

Here is a link to a doc which explains what I did, along with some comments / questions about how to potentially improve the user experience, as well as a few possible bugs. 

https://docs.google.com/document/d/1OQxCo1ijFKlV3-A5vYkL3NtK-ulBfY28oPglxE6JJco/edit?usp=sharing

I can make them into issues if you want me to but didn't want to do that until  you had a chance to look through the doc.
